bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -148,14 +166,21 @@ public String getId() {
         }
     }
 
+    @JsonIgnore
+    @Nullable
+    public String getRedirectUrl() {
+        return GuideUtils.getRedirectUrl(redirect, getPath());
+    }
+
+    @JsonIgnore
+    @Nullable
+    public String getPrefillService() {
+        return prefillService;
+    }
+
     @Override
     public String getAction() {
-        if (getCurrentPage() != null) {
-            return ADOBE_GLOBAL_API_ROOT + FORMS_RUNTIME_API_GLOBAL_ROOT + "/submit/" + ComponentUtils.getEncodedPath(getCurrentPage()
-                .getPath());
-        } else {
-            return null;
-        }
+        return ADOBE_GLOBAL_API_ROOT + FORMS_RUNTIME_API_GLOBAL_ROOT + "/submit/" + ComponentUtils.getEncodedPath(getPath());
```

Comment:
we cannot merge this as it'll break the next release. pls have a logic to deduce if form has been authored from sites editor.

Reply:
added a check in getAction() to check if forms created in sites